### PR TITLE
soc: riscv: nordic_nrf: vpr: enable XIP capability

### PR DESCRIPTION
**Pull request by gmarull**
This PR allows using CONFIG_XIP=y for the VPR core. Also:

- Update nRF54H20PDK cpuppr board to use XIP by default
- Add new snippets for execution from RAM, so behavior can be easily changed

depends on https://github.com/zephyrproject-rtos/zephyr/pull/69477

### CODE REVIEW

File: snippets/nordic-ppr-ram/nordic-ppr-ram.overlay [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* code partition size must match RAM size */
+&cpuppr_code_partition {
+	reg = <0x126000 DT_SIZE_K(28)>;
+};
```
Comment (57300) [inline snippets/nordic-ppr-ram/nordic-ppr-ram.overlay:9]: This adjustment isn't required when building for PPR, only for its owner.

File: soc/riscv/nordic_nrf/common/vpr/Kconfig [hunk 2]
```diff
@@ -14,5 +14,6 @@ config RISCV_CORE_NORDIC_VPR
 	select RISCV_ISA_EXT_ZIFENCEI
 	select RISCV_SOC_HAS_ISR_STACKING
 	select RISCV_SOC_CONTEXT_SAVE
+	select HAS_FLASH_LOAD_OFFSET if XIP
```
Comment (57300) [inline soc/riscv/nordic_nrf/common/vpr/Kconfig:17]: Not sure that it needs to be `if XIP`.

Comment (gmarull) [reply soc/riscv/nordic_nrf/common/vpr/Kconfig:17]: I guess it doesn't hurt if always enabled?

File: snippets/nordic-ppr-ram/nordic-ppr-ram.overlay (hunk 1 above)
Comment (gmarull) [reply snippets/nordic-ppr-ram/nordic-ppr-ram.overlay:15]: good catch, actually dropped the snipped as it was just CONFIG_XIP setting...

**Review by 57300 — DISMISSED** [review]

File: snippets/nordic-ppr-ram/README.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.. _nordic-boot-ppr-ram:
+
+Nordic boot PPR snippet with execution from RAM (nordic-boot-ppr-ram)
+#####################################################################
+
+Overview
+********
+
+This snippet allows users to build Zephyr with the capability to boot Nordic PPR
+(Peripheral Processor) from another core. PPR code is executed executed from
```
Comment (anangl) [inline snippets/nordic-ppr-ram/README.rst:10]:
```suggestion
(Peripheral Processor) from another core. PPR code is executed from
```

File: snippets/nordic-ppr-ram/README.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
+.. _nordic-boot-ppr-ram:
+
+Nordic boot PPR snippet with execution from RAM (nordic-boot-ppr-ram)
```
Comment (anangl) [inline snippets/nordic-ppr-ram/README.rst:3]: Shouldn't this match the actual snippet name, i.e. `nordic-ppr-ram`?

**Review by anangl — DISMISSED** [review]
Looks good. Two nit-picking comments only.

**Review by anangl — APPROVED** [review]

**Review by 57300 — APPROVED** [review]

**Pull request merged**